src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,84 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.model.InvalidDataException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_LENGTH = 6;
+    private static final int MIN_LOGIN_LENGTH = 3;
+    private static final int MIN_AGE = 18;
+    private static final int ZERO_AGE = 0;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
-    public User register(User user) {
-        return null;
+    public User register(User user) throws InvalidDataException {
+        userValidation(user);
+        passwordValidation(user);
+        ageValidation(user);
+        loginValidation(user);
+        return storageDao.add(user);
+    }
+
+    private void userValidation(User user) throws InvalidDataException {
+        if (user == null) {
+            throw new InvalidDataException("The user should not be equal null");
+        }
+
+        User newUser = storageDao.get(user.getLogin());
+        if (newUser != null) {
+            throw new InvalidDataException("The user already is in storage");
+        }
+    }
+
+    private void passwordValidation(User user) throws InvalidDataException {
+        if (user.getPassword() == null) {
+            throw new InvalidDataException("Your password is equal null"
+                    + user.getPassword());
```

Comment:
Here you don't need to add `user.getPassword()` as it is null

Reply:
The same for all messages after null checks

Reply:
Thank you for your good feadback :)